### ISSUE TITLE
MiZiP keys and card read support

### ISSUE THREAD

**Issue by LowSkillDeveloper** [flipperdevices/flipperzero-firmware#2385]
I have a "MiZiP" key for the coffee machine. As I understand it, they come in 2 types, NFC and RFID. And maybe they contain information about the balance on the key. But the flipper can't read the balance from these keys.
Or maybe they use a database on their side, . But even so, flipper cannot read the card id.

![MIZIP](https://user-images.githubusercontent.com/25121341/218278516-43397a4d-f52e-40a8-96e9-dcd3805be96d.jpg)

My flipper and phone does not see the NFC, so I can assume that the key is RFID.
When reading RFID with Flipper Zero, it gives this:

![Screenshot-20230211-205306](https://user-images.githubusercontent.com/25121341/218278492-de5cb420-1a22-490c-9544-1809774602e8.png)

> Filetype: Flipper RFID key
> Version: 1
> Key type: EM4100
> Data: 00 00 00 00 00

I also did "Read RAW RFID data" when the balance on the key was different. 88 coins, 82, 78, 66.
I don't know how and which program use to read RAW files, so I'll just attach them here:
https://github.com/LowSkillDeveloper/mizip-flipper-temp

Having a dump of the same key with different balances, I think it is possible to find a changing value if it is in the key. Or at least Flipper will be able to read the id of the key.

Also this is official site with information about MiZiP: https://newis.evocagroup.com/en/payment-solutions/mizip
Perhaps they also use specific NFC, so the flipper can't detect it?

**Comment by augustozanellato** [comment]
Mizip stands for Mifare Zip, it’s a mifare mini tag. 
Lf-rfid ZIP keys are based on pcf7931 which flipper doesn’t support due to hw limitations iirc

**Comment by skotopes** [comment]
@augustozanellato theoretically pcf7931 can be implemented on top of flipper hardware. At least 5577 is working fine. Question is how stable it will be.

**Comment by augustozanellato** [comment]
> @augustozanellato theoretically pcf7931 can be implemented on top of flipper hardware. At least 5577 is working fine. Question is how stable it will be.

Oh, so read only would be feasible? Last time I asked about PCF transponder in discord I was told that it wouldn’t be possible due to hw limitations. 
cfr https://discord.com/channels/740930220399525928/954422698879098990/1040213253915156560

**Comment by skotopes** [comment]
Yep, there are some challenges described by @DrZlo13 ;-)

**Comment by augustozanellato** [comment]
> Yep, there are some challenges described by @DrZlo13 ;-)

But I guess you like challenges😂



more info on MiZIP btw: 
Key derivation from uid: https://github.com/RfidResearchGroup/proxmark3/blob/master/client/luascripts/hf_mf_uidkeycalc_mizip.lua
Data encoding: https://slides.com/siras/chibre/fullscreen

**Comment by LowSkillDeveloper** [comment]
Are there any updates?

**Comment by nocturn9x** [comment]
I have one of those and I can read it with my smartphone just fine. Maybe there's both NFC and RFID models?

**Comment by LowSkillDeveloper** [comment]
> Maybe there's both NFC and RFID models?

Yes, as I understand there are 2 versions. But I wonder how the data saving works in the rfid version... Of course, there is an option that there is no data there, and everything is on the server in the database and a simple id binding. But as we can see, the id on my photo and other data are zero, but the keys works.

**Comment by nocturn9x** [comment]
I'm not familiar with RFID devices, so I can't say. I've messed a bunch with MiZip NFC keys though, I should dig mine back up actually.

But yeah, if it's RFID then 🤷
